python_modules/libraries/dagster-dbt/dagster_dbt/cloud/asset_defs.py

```diff
@@ -75,10 +75,9 @@ def __init__(
             if isinstance(dbt_cloud_resource_def, DbtCloudClientResource)
             else dbt_cloud_resource_def
         )
+
         self._dbt_cloud: DbtCloudClient = (
-            dbt_cloud_resource_def.with_resource_context(
-                build_init_resource_context()
-            ).get_dbt_client()  # type: ignore
+            dbt_cloud_resource_def._process_config_and_initialize().get_dbt_client()  # noqa: SLF001
```

Comment:
I am surprised that the user would need to use a private method here to get the underlying object from a resource  definition.

1. Can this method be public?
2. Looks like the `.with_resource_context` and `.from_resource_context` methods are confusing right now, given that we've made this mistake. Could we potentially consolidate them or rename them?